src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -354,36 +354,39 @@ public ReadPileup getOverlappingFragmentFilteredPileup() {
      *
      * @return the newly filtered pileup
      */
-    public ReadPileup getOverlappingFragmentFilteredPileup(boolean discardDiscordant, Comparator<PileupElement> tieBreaker) {
-        Map<String, PileupElement> filteredPileup = new HashMap<String, PileupElement>();
-        Set<String> readNamesDeleted = new HashSet<>();
+    public ReadPileup getOverlappingFragmentFilteredPileup(boolean discardDiscordant, Comparator<PileupElement> tieBreaker, SAMFileHeader header) {
+        List<PileupElement> filteredPileupList = new ArrayList<PileupElement>();
 
-        for (PileupElement p : this) {
-            String readName = p.getRead().getName();
+        for( Map.Entry<String, ReadPileup> pileup : this.splitBySample(header, null).entrySet()) {
+            Map<String, PileupElement> filteredPileup = new HashMap<String, PileupElement>();
+            Set<String> readNamesDeleted = new HashSet<>();
 
-            // if we've never seen this read before, life is good
-            if (!filteredPileup.containsKey(readName)) {
-                if(!readNamesDeleted.contains(readName)) {
-                    filteredPileup.put(readName, p);
-                }
-            } else {
-                PileupElement existing = filteredPileup.get(readName);
+            for (PileupElement p : pileup.getValue()) {
+                String readName = p.getRead().getName();
 
-                // if the reads disagree at this position, throw them all out.  Otherwise
-                // keep the element with the highest quality score
-                if (discardDiscordant && existing.getBase() != p.getBase()) {
-                    filteredPileup.remove(readName);
-                    readNamesDeleted.add(readName);
-                } else {
-                    if (tieBreaker.compare(existing, p) < 0) {
+                // if we've never seen this read before, life is good
+                if (!filteredPileup.containsKey(readName)) {
+                    if(!readNamesDeleted.contains(readName)) {
                         filteredPileup.put(readName, p);
                     }
+                } else {
+                    PileupElement existing = filteredPileup.get(readName);
+
+                    // if the reads disagree at this position, throw them all out.  Otherwise
+                    // keep the element with the highest quality score
+                    if (discardDiscordant && existing.getBase() != p.getBase()) {
+                        filteredPileup.remove(readName);
+                        readNamesDeleted.add(readName);
+                    } else {
+                        if (tieBreaker.compare(existing, p) < 0) {
```

Comment:
I think this if statement should be moved up into the else as an `else if(..)`.  Less nesting is usually better if it's equivalent which this is.